Add headers to KNOWN_ISSUES (#53)
IBM/cloudant-go-sdk: KNOWN_ISSUES.md

  ```
  json: cannot unmarshal string into Go value of type map[string]json.RawMessage
  ```

### Bodies containing the `headers` parameter

> The `Headers` map is reserved for the API request headers.

Reviewer: ```suggestion
The `Headers` field is always reserved for the request headers.
```

Author: I was brave enough to say it is a map, because the Go-wrapped headers type is a map. I also used `API request headers` because the comment above each `Headers` says the same. E.g:
https://github.com/IBM/cloudant-go-sdk/blob/fc0ed4d90c1ef0c8ee7f019455a3d47decbd7ad6/cloudantv1/cloudant_v1.go#L9110-L9111

In order to avoid collision between the request headers and the `headers` parameter

> that is contained by the request body, use the `HeadersVar` as
> the request body `headers` parameter.

Reviewer: ```suggestion
that is contained by `ReplicationDatabase` in the request body, it has a different field name: `HeadersVar`.
```

E.g. for applying the `headers` parameter of your replicate
target database, use the following approach:

```go
...
target, _ := cloudant.NewReplicationDatabase(
	"<your-service-url>" + "/" + "animaldb-target",
)

target.HeadersVar = make(map[string]string)

> target.HeadersVar["Authorization"] = "<your-base64-encoded-auth-key>"

Reviewer: ```suggestion
target.HeadersVar["Authorization"] = "Basic <your-base64-encoded-auth-key>"
```
Here is a documentation about Basic authentication form: https://en.wikipedia.org/wiki/Basic_access_authentication

Author: Fix in 4a204ff.

)
...
```
The example above represents this JSON body:
```json
{
    ...
    "target": {
        "headers": {

>             "Authorization": "<your-base64-encoded-auth-key>"

Reviewer: ```suggestion
            "Authorization": "Basic <your-base64-encoded-auth-key>"
```

Author: Fix in 4a204ff.
